jax/_src/third_party/scipy/interpolate.py

```diff
@@ -0,0 +1,155 @@
+import scipy.interpolate as osp_interpolate
+
+from jax._src.tree_util import register_pytree_node
+from jax._src.numpy.lax_numpy import (_promote_dtypes_inexact, asarray,
+                                      broadcast_arrays, can_cast, empty, nan,
+                                      searchsorted, where, zeros)
+from jax._src.numpy.util import _wraps
+
+
+def _ndim_coords_from_arrays(points, ndim=None):
+  """Convert a tuple of coordinate arrays to a (..., ndim)-shaped array."""
+  if isinstance(points, tuple) and len(points) == 1:
+    # handle argument tuple
+    points = points[0]
+  if isinstance(points, tuple):
+    p = broadcast_arrays(*points)
+    for p_other in p[1:]:
+      if p_other.shape != p[0].shape:
+        raise ValueError("coordinate arrays do not have the same shape")
+    points = empty(p[0].shape + (len(points),), dtype=float)
+    for j, item in enumerate(p):
+      points = points.at[..., j].set(item)
+  else:
+    points = asarray(points)  # SciPy: asanyarray(points)
```

Comment:
Please call `_check_arraylike()` on `points` before `asarray`

Reply:
Will do!

Just out of curiosity since I wasn't able to quickly find the reason in the code: Is this for better error messages or are there cases where `asarray` works but does not yield the expected array?

Reply:
It's mainly to prevent silent slowdowns if users pass Python sequences to JIT-compiled functions; some context in #7737